RockPaperScissors/RockPaperScissors/GameLogic/RPSGameRule.swift

```diff
@@ -0,0 +1,58 @@
+//
+//  RPSGameRule.swift
+//  RockPaperScissors
+//
+//  Created by Park Sungmin on 3/19/24.
+//
+
+import Foundation
+
+protocol GameRule {
+    var score: GameScore { get }
+    var gameState: GameState { get }
+    func match(myAction: Actions, opponentAction: Actions) -> MatchResult
+}
+
+final class RPSGameRule: GameRule {
+    init(targetScore: Int) {
+        score = GameScore(targetScore: targetScore)
+        gameState = .playing
+    }
+    
+    var score: GameScore
+    var gameState: GameState
+    
+    func match(myAction: Actions, opponentAction: Actions) -> MatchResult {
+        let result = (myAction.rawValue - opponentAction.rawValue + 3) % 3
+        return MatchResult(rawValue: result)!
+    }
```

Comment:
논리의 흐름이 Actions -> Actions.rawValue(Int) -> MatchResult.rawValue(Int) -> MatchResult 이런 모양으로 흘러가는 데요.
개인적으로는 enum의 rawValue 보다 구체화된 타입을 그대로 사용하는 것을 선호해요.
구현하는 방식에는 여러가지가 있겠지만 이런 방법도 있다는 것을 참고 해주시면 좋을 것 같아요.

```suggestion
    func match(myAction: Actions, opponentAction: Actions) -> MatchResult {
      switch myAction, opponentAction {
        case .rock, .rock: return .draw
        ...
      }  
    }
```

Reply:
이번에 코드 작성하면서 꽤나 기획 변경에 따른 코드 수정이 가능하도록 코드를 작성하려 노력했다고 생각했지만, 
막상 STEP 2를 받아드니 전체적인 코드의 수정이 가해져야 하는 것 같아서 당황스럽게 느껴졌던 것 같아요 ㅋㅋㅋ

사실 제임히가 제안해주신 것 처럼 Switch에 여러 변수를 물려서 case로 구분하는 방법은 전혀 생각하지 못했어요! 
`switch myAction { switch opponentAction { ... } }` 처럼 2중으로 switch를 해야하나? 라는 생각을 하다가 아예 한 줄로 끝내버리자는 의도로 수학으로 풀어버렸는데, 묵찌빠로의 룰 확장 등을 생각한다면 제안해주신 방법처럼 좀 더 가독성을 가지고 코드에서 규칙을 보여줄 수 있는 방향으로 작성하는게 좋을 것 같다고 생각이 됩니다.